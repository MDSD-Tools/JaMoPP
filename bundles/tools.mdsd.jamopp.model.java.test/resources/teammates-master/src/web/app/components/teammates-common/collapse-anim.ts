import { animate, AnimationTriggerMetadata, style, transition, trigger } from '@angular/animations';

/**
 * Directive for collapsing of *ngIf columns.
 */
export const collapseAnim: AnimationTriggerMetadata = trigger('collapseAnim', [
  transition(':leave', [
    style({ height: '*', overflow: 'hidden' }),
    animate('300ms ease-in-out', style({ height: 0, opacity: 0 })),
  ]),
  transition(':enter', [
    style({ height: '0', overflow: 'hidden' }),
    animate('300ms ease-in-out', style({ height: '*', opacity: 1 })),
  ]),
]);
