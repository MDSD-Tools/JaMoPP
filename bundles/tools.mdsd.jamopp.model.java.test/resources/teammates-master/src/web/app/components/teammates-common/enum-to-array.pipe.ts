import { Pipe, PipeTransform } from '@angular/core';

/**
 * Pipe to handle the transformation of an enum to an array of all types it has.
 *
 * <p>Assums the enum is string in runtime.
 */
@Pipe({
  name: 'enumToArray',
})
export class EnumToArrayPipe implements PipeTransform {

  /**
   * Transforms enum to an array of all types it has.
   */
  transform(enumObj: any): any[] {
    return Object.keys(enumObj).map((s: string) => enumObj[s]);
  }

}
