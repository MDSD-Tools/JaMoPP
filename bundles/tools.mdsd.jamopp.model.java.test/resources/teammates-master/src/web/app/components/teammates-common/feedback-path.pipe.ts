import { Pipe, PipeTransform } from '@angular/core';
import { FeedbackParticipantType } from '../../../types/api-output';

/**
 * Pipe to handle the display of {@link FeedbackParticipantType} from giver's perspective.
 */
@Pipe({
  name: 'giverTypeDescription',
})
export class GiverTypeDescriptionPipe implements PipeTransform {

  /**
   * Transforms {@link FeedbackParticipantType} to a simple description from giver's perspective.
   */
  transform(type: FeedbackParticipantType): any {
    switch (type) {
      case FeedbackParticipantType.SELF:
        return 'Feedback session creator (i.e., me)';
      case FeedbackParticipantType.STUDENTS:
        return 'Students in this course';
      case FeedbackParticipantType.INSTRUCTORS:
        return 'Instructors in this course';
      case FeedbackParticipantType.TEAMS:
        return 'Teams in this course';
      default:
        return 'Unknown';
    }
  }

}

/**
 * Pipe to handle the display of {@link FeedbackParticipantType} from recipient's perspective.
 */
@Pipe({
  name: 'recipientTypeDescription',
})
export class RecipientTypeDescriptionPipe implements PipeTransform {

  /**
   * Transforms {@link FeedbackParticipantType} to a simple description from recipient's perspective.
   */
  transform(type: FeedbackParticipantType): any {
    switch (type) {
      case FeedbackParticipantType.SELF:
        return 'Giver (Self feedback)';
      case FeedbackParticipantType.STUDENTS:
        return 'Other students in the course';
      case FeedbackParticipantType.INSTRUCTORS:
        return 'Instructors in the course';
      case FeedbackParticipantType.TEAMS:
        return 'Other teams in the course';
      case FeedbackParticipantType.OWN_TEAM:
        return "Giver's team";
      case FeedbackParticipantType.OWN_TEAM_MEMBERS:
        return "Giver's team members";
      case FeedbackParticipantType.OWN_TEAM_MEMBERS_INCLUDING_SELF:
        return "Giver's team members and Giver";
      case FeedbackParticipantType.NONE:
        return 'Nobody specific (For general class feedback)';
      default:
        return 'Unknown';
    }
  }

}
