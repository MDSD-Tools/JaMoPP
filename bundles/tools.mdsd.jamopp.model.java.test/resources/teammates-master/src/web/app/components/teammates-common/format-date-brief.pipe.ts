import { Pipe, PipeTransform } from '@angular/core';
import { TimezoneService } from '../../../services/timezone.service';

/**
 * Pipe to handle the display of feedback session start and end time in brief.
 */
@Pipe({
  name: 'formatDateBrief',
})
export class FormatDateBriefPipe implements PipeTransform {

  constructor(private timezoneService: TimezoneService) {}

  /**
   * Transforms a timestamp to a date string briefly.
   */
  transform(timestamp: number, timeZone: string): string {
    return this.timezoneService.formatToString(timestamp, timeZone, 'D MMM h:mm A');
  }

}
