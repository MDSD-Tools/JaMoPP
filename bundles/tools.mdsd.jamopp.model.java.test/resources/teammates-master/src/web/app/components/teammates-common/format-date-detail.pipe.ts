import { Pipe, PipeTransform } from '@angular/core';
import { TimezoneService } from '../../../services/timezone.service';

/**
 * Pipe to handle the display of a timestamp in detail.
 */
@Pipe({
  name: 'formatDateDetail',
})
export class FormatDateDetailPipe implements PipeTransform {

  constructor(private timezoneService: TimezoneService) {}

  /**
   * Transforms a timestamp to a date string in detail.
   */
  transform(timestamp: number, timeZone: string): string {
    return this.timezoneService.formatToString(timestamp, timeZone, 'ddd, DD MMM YYYY, HH:mm A z');
  }

}
