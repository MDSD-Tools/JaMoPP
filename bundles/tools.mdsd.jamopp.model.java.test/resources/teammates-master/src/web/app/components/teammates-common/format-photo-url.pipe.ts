import { Pipe, PipeTransform } from '@angular/core';

import { environment } from '../../../environments/environment';

/**
 * Pipe to handle formatting of URL for API request of photo retrieval
 */
@Pipe({
  name: 'formatPhotoUrl',
})
export class FormatPhotoUrlPipe implements PipeTransform {

  transform(email: string, courseId: string): string {
    return `${environment.backendUrl}/webapi/student/profilePic?courseid=${courseId}&studentemail=${email}`;
  }
}
