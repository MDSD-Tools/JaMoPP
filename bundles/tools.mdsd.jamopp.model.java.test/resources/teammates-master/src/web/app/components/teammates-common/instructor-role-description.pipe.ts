import { Pipe, PipeTransform } from '@angular/core';
import { InstructorPermissionRole } from '../../../types/api-output';

/**
 * Pipe to handle the transformation of an InstructorPermissionRole to a description.
 */
@Pipe({
  name: 'instructorRoleDescription',
})
export class InstructorRoleDescriptionPipe implements PipeTransform {

  /**
   * Transforms InstructorPermissionRole to a description.
   */
  transform(role: InstructorPermissionRole): any {
    switch (role) {
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER:
        return 'Co-owner: Can do everything.';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_MANAGER:
        return 'Manager: Can do everything except for deleting/restoring the course.';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_OBSERVER:
        return 'Observer: Can only view information(students, submissions, comments etc.). '
            + 'Cannot edit/delete/submit anything.';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_TUTOR:
        return 'Tutor: Can view student details, give/view comments, submit/view responses for sessions.';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_CUSTOM:
      default:
        return 'Custom: No access by default. Any access needs to be granted explicitly.';
    }
  }

}
