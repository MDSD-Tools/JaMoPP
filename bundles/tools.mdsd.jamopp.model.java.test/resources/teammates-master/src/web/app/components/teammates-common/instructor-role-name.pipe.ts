import { Pipe, PipeTransform } from '@angular/core';
import { InstructorPermissionRole } from '../../../types/api-output';

/**
 * Pipe to handle the transformation of an InstructorPermissionRole to a name.
 */
@Pipe({
  name: 'instructorRoleName',
})
export class InstructorRoleNamePipe implements PipeTransform {

  /**
   * Transforms InstructorPermissionRole to a name.
   */
  transform(role: InstructorPermissionRole): any {
    switch (role) {
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER:
        return 'Co-owner';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_MANAGER:
        return 'Manager';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_OBSERVER:
        return 'Observer';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_TUTOR:
        return 'Tutor';
      case InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_CUSTOM:
      default:
        return 'Custom';
    }
  }

}
