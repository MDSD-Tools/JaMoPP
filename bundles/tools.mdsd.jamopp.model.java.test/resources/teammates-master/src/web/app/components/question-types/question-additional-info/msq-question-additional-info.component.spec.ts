import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import {
  FeedbackMsqQuestionDetails,
  FeedbackParticipantType,
  FeedbackQuestionType,
} from '../../../../types/api-output';
import { NO_VALUE } from '../../../../types/feedback-response-details';
import { TeammatesCommonModule } from '../../teammates-common/teammates-common.module';
import { MsqQuestionAdditionalInfoComponent } from './msq-question-additional-info.component';

describe('MsqQuestionAdditionalInfoComponent', () => {
  let component: MsqQuestionAdditionalInfoComponent;
  let fixture: ComponentFixture<MsqQuestionAdditionalInfoComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [MsqQuestionAdditionalInfoComponent],
      imports: [TeammatesCommonModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MsqQuestionAdditionalInfoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  const msqDetails: FeedbackMsqQuestionDetails = {
    msqChoices: ['1', '2', '3'],
    otherEnabled: false,
    generateOptionsFor: FeedbackParticipantType.NONE,
    maxSelectableChoices: NO_VALUE,
    minSelectableChoices: NO_VALUE,
    hasAssignedWeights: false,
    msqWeights: [],
    msqOtherWeight: 0,
    questionType: FeedbackQuestionType.MSQ,
    questionText: 'MSQ question detail',
  };

  const msqDetailsWithGeneratedFor: FeedbackMsqQuestionDetails = {
    msqChoices: ['1', '2', '3'],
    otherEnabled: false,
    generateOptionsFor: FeedbackParticipantType.STUDENTS,
    maxSelectableChoices: NO_VALUE,
    minSelectableChoices: NO_VALUE,
    hasAssignedWeights: false,
    msqWeights: [],
    msqOtherWeight: 0,
    questionType: FeedbackQuestionType.MSQ,
    questionText: 'MSQ question detail',
  };

  const msqDetailsWithOther: FeedbackMsqQuestionDetails = {
    msqChoices: ['1', '2', '3'],
    otherEnabled: true,
    generateOptionsFor: FeedbackParticipantType.NONE,
    maxSelectableChoices: NO_VALUE,
    minSelectableChoices: NO_VALUE,
    hasAssignedWeights: false,
    msqWeights: [],
    msqOtherWeight: 0,
    questionType: FeedbackQuestionType.MSQ,
    questionText: 'MSQ question detail',
  };

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default view', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should show msq options', () => {
    component.questionDetails = msqDetails;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show other when "other" option is enabled', () => {
    component.questionDetails = msqDetailsWithOther;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should show generated option message when generateOptionFor is enabled', () => {
    component.questionDetails = msqDetailsWithGeneratedFor;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
