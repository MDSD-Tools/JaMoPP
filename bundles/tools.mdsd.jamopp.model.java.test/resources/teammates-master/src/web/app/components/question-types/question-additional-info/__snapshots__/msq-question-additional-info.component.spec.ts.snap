// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MsqQuestionAdditionalInfoComponent should show generated option message when generateOptionFor is enabled 1`] = `
<tm-msq-question-additional-info
  FeedbackParticipantType={[Function Object]}
  questionDetails={[Function Object]}
>
  Multiple-choice (multiple answers) question options:
  <div>
     The options for this question is automatically generated from the list of all students in this course.
  
  </div>
</tm-msq-question-additional-info>
`;

exports[`MsqQuestionAdditionalInfoComponent should show msq options 1`] = `
<tm-msq-question-additional-info
  FeedbackParticipantType={[Function Object]}
  questionDetails={[Function Object]}
>
  Multiple-choice (multiple answers) question options:
  <ul>
    
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3
    </li>
    
  </ul>
</tm-msq-question-additional-info>
`;

exports[`MsqQuestionAdditionalInfoComponent should show other when "other" option is enabled 1`] = `
<tm-msq-question-additional-info
  FeedbackParticipantType={[Function Object]}
  questionDetails={[Function Object]}
>
  Multiple-choice (multiple answers) question options:
  <ul>
    
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3
    </li>
    
    <li>
      Other
    </li>
  </ul>
</tm-msq-question-additional-info>
`;

exports[`MsqQuestionAdditionalInfoComponent should snap with default view 1`] = `
<tm-msq-question-additional-info
  FeedbackParticipantType={[Function Object]}
  questionDetails={[Function Object]}
>
  Multiple-choice (multiple answers) question options:
  <ul>
    
    
  </ul>
</tm-msq-question-additional-info>
`;
